sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -383,6 +384,62 @@ def test_py_volume(self):
     """Test a pipeline with a volume and volume mount."""
     self._test_py_compile_yaml('volume')
 
+  def test_py_compile_mode(self):
+    cases = [
+      {'mode': 'V2_COMPATIBLE', 'is_v2': True},
+      {'mode': 'V1', 'is_v2': False},
+      {'mode': 'V1_LEGACY', 'is_v2': False},
+      {'mode': None, 'is_v2': False},
+      {'mode': 'V2_COMPATIBLE', 'env': 'V1', 'is_v2': True},
+      {'mode': None, 'env': 'V1', 'is_v2': False},
+      {'mode': None, 'env': 'V2_COMPATIBLE', 'is_v2': True},
+      {'mode': None, 'env': 'V1_LEGACY', 'is_v2': False},
+      {'mode': 'INVALID', 'error': True},
+      {'mode': None, 'env': 'INVALID', 'error': True},
```

Comment:
Hi @chensun, I'm adding a table driven test, because this feels like a right fit for full coverage.
How do you feel about this?

Reply:
A more standard way to do the same would be using parameterized tests: https://github.com/abseil/abseil-py/blob/master/absl/testing/parameterized.py

We also have examples in our existing UTs. For example:
https://github.com/kubeflow/pipelines/blob/2abc352ad8617afb1e299360a5b2a5f9da5772ae/sdk/python/kfp/dsl/type_utils_test.py#L139

Reply:
Thanks, updated to absl.testing.parameterized!
There's an obvious improvement on error message